crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -582,20 +573,25 @@ impl LogletConfiguration {
     fn try_reconfiguring(
         &self,
         log_id: LogId,
+        last_segment_index: SegmentIndex,
         logs_configuration: &LogsConfiguration,
         observed_cluster_state: &ObservedClusterState,
         preferred_sequencer: Option<NodeId>,
     ) -> Option<LogletConfiguration> {
-        let loglet_id = self.loglet_id();
+        let next_loglet_id = match self {
+            Self::Local(_) => LogletId::new(log_id, last_segment_index.next()),
+            #[cfg(any(test, feature = "memory-loglet"))]
+            Self::Memory(_) => LogletId::new(log_id, last_segment_index.next()),
+            #[cfg(feature = "replicated-loglet")]
+            Self::Replicated(params) => params.loglet_id.next(),
```

Comment:
Why not (log_id, last_segment_index.next()) here as well?

Reply:
I had the impression that we should always use loglet_id.next() of a replicated loglet. And only rely on the segment index iff we migrating from local/memory loglet. What made me not always use the segment index is:
- Why do we have a loglet index if we can always just use the log_id + segment_index?
- in case a segment was configured manually with a loglet id that is out of sync, that the next segment doesn't go backward

Reply:
> Why do we have a loglet index if we can always just use the log_id + segment_index?

I'm not sure what loglet index is.

> in case a segment was configured manually with a loglet id that is out of sync, that the next segment doesn't go backward

I don't think we let the user supply the loglet is in "restatectl reconfigure" command, is there other places where we might allow loglet-ids (for replicated loglet) to be arbitrary values supplied by the user?

Reply:
> I'm not sure what loglet index is.
I meant the loglet id

>I don't think we let the user supply the loglet is in "restatectl reconfigure" command, is there other places where we might allow loglet-ids (for replicated loglet) to be arbitrary values supplied by the user?

True, but the grpc API allows it. I talked with @tillrohrmann about it and We decided:
- Always use log_id+segment index
- Modify the seal_and_extend api to be more restrictive and does not accept LogParams anymore (follow up)

Reply:
loglet-ids are up to the log provider, what we are deciding here is for loglets that we own, we provider consistent convention for generating them. Other loglet providers can have arbitrary values.